### PR TITLE
Preserve encoded HTML characters

### LINKED ISSUE #6: Retain encoded HTML characters
Premailer erroneously changes encoded HTML characters into ASCII. For example `&rsquo;` is converted to an ASCII single quote `'`. The encoded characters should be retained in either an alphanumeric or unicode value.

### PR DESCRIPTION
**Pull request by shirgoldbird**
Fixes #6

### CODE REVIEW

**Comment by pennetti** [comment]
Could you add a spec for this please?

**Comment by shirgoldbird** [comment]
@pennetti Added!

File: spec/fixtures/text/with-encoded-chars.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+&#8217;
```
Comment (jeremyckahn) [inline spec/fixtures/text/with-encoded-chars.erb]: I don't believe that text HTMLs should have HTML entities, correct?

File: lib/getinline/transformer.rb [hunk 2]
```diff
@@ -22,8 +23,8 @@ def transform
 
       @premailer = Premailer.new(TOKENIZED_ERB_FILE_NAME, @premailer_options)
       premailed_tokenized_text = @options[:mode] == :txt ?
-        @premailer.to_plain_text : @premailer.to_inline_css
-      premailed_text = premailed_tokenized_text.dup
+        Nokogiri::HTML.fragment(@premailer.to_plain_text) : Nokogiri::HTML(@premailer.to_inline_css)
```
Comment (jeremyckahn) [inline lib/getinline/transformer.rb]: I believe that HTML encoding text email content is not what we want.

Comment (shirgoldbird) [inline lib/getinline/transformer.rb]: @jeremyckahn Derp- what was I thinking? Removed.

File: spec/getinline/transformer_spec.rb [hunk 3]
```diff
@@ -45,5 +45,27 @@
...
+    context 'with special characters' do
+      let(:raw_with_special_characters) { File.read('spec/fixtures/raw/with-encoded-chars.erb') }
+      let(:text_with_special_characters) { File.read('spec/fixtures/text/with-encoded-chars.erb') }
+      let(:inlined_with_special_characters) { File.read('spec/fixtures/inlined/with-encoded-chars.erb') }
+
+      context 'with html tags' do
+        let(:transformer) { described_class.new(raw_with_special_characters) }
+        let(:transformed_html) { transformer.transform }
+        it 'encodes special characters to ASCII' do
+          expect(transformed_html).to include(inlined_with_special_characters)
```
Comment (jabbowaki) [inline spec/getinline/transformer_spec.rb:58]: heeeey. When Jeremy originally wrote the specs, Nicholas suggested removing the `tranformed_html`let and calling it in the spec itself. Something about how lets aren't really supposed to be doing logic or actions or something like that. So instead it'd be something like:
`expect(transformer.transform).to include(inlined_with_special_characters)`
It's a total nitpick though.

File: spec/getinline/transformer_spec.rb [hunk 4]
```diff
@@ -45,5 +45,27 @@
         end
       end
     end
+
+    context 'with special characters' do
```
Comment (pennetti) [inline spec/getinline/transformer_spec.rb:49]: We may want to refactor these down the line so we can test templates with both ERB tags and special characters - you asked about this before and I misunderstood the question.

File: lib/getinline/transformer.rb [hunk 5]
```diff
@@ -22,7 +23,7 @@ def transform
 
       @premailer = Premailer.new(TOKENIZED_ERB_FILE_NAME, @premailer_options)
       premailed_tokenized_text = @options[:mode] == :txt ?
-        @premailer.to_plain_text : @premailer.to_inline_css
+        @premailer.to_plain_text : Nokogiri::HTML(@premailer.to_inline_css).to_html(encoding:'US-ASCII')
```
Comment (dispatchrabbi) [inline lib/getinline/transformer.rb:26]: Why ASCII and not UTF-8? Is it an Outlook 1879 thing?

Comment (shirgoldbird) [inline lib/getinline/transformer.rb:26]: @dispatchrabbi ~~Yes~~ Probably.

**Pull request merged**